connect/src/com/telenor/connect/sms/AppSignatureHelper.java

```diff
@@ -0,0 +1,80 @@
+package com.telenor.connect.sms;
+
+import android.content.Context;
+import android.content.ContextWrapper;
+import android.content.pm.PackageManager;
+import android.content.pm.Signature;
+import android.util.Base64;
+import android.util.Log;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * This is a helper class to generate your message hash to be included in your SMS message.
+ *
+ * Without the correct hash, your app won't recieve the message callback. This only needs to be
+ * generated once per app and stored. Then you can remove this helper class from your code.
+ */
+public class AppSignatureHelper extends ContextWrapper {
+    public static final String TAG = AppSignatureHelper.class.getSimpleName();
+
+    private static final String HASH_TYPE = "SHA-256";
+    public static final int NUM_HASHED_BYTES = 9;
+    public static final int NUM_BASE64_CHAR = 11;
+
+    public AppSignatureHelper(Context context) {
+        super(context);
+    }
+
+    /**
+     * Get all the app signatures for the current package
+     * @return
+     */
+    public ArrayList<String> getAppSignatures() {
```

Comment:
I usually like return types and variable types of collection classes to be the interface rather than the concrete implementation; unless you rely specifically on the implementation where the value is returned to.

Reply:
Agree, but this is actual borrowed Google code: https://github.com/googlesamples/android-credentials/blob/master/sms-verification/android/app/src/main/java/com/google/samples/smartlock/sms_verify/AppSignatureHelper.java

I think perhaps I'm leaning towards just deleting this file, and just referring to the Google documentation, in our docs (https://developers.google.com/identity/sms-retriever/verify#computing_your_apps_hash_string). The code should never be in any production product anyways.